### PR TITLE
Bump scikit-learn version to 1.3.0

### LINKED ISSUE #3050: scikit-learn 1.0.2 does not have wheels for python 3.11
/kind bug

**What steps did you take and what happened:**
sklearn servingruntime tests are failing in python 3.11. The scikit-learn 1.0.2 does not support python 3.11. During installation scikit-learn is compiled from the source. I believe it has some incompatibility with the cython version. From version 1.2.0 scikit learn provides wheels for python 3.11

**What did you expect to happen:**

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version:
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:
sklearn servingruntime tests are failing in python 3.11. The scikit-learn 1.0.2 does not support python 3.11. During installation scikit-learn is compiled from the source. I believe it has some incompatibility with the cython version. From version 1.2.0 scikit learn provides wheels for python 3.11 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3050 

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: python/sklearnserver/pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "sklearnserver"
-version = "0.11.0rc0"
+version = "0.11.0rc1"
```
Comment (yuzisun) [inline python/sklearnserver/pyproject.toml:3]: Oh we need to update other places too ? Need to run Andrew’s script I think

Comment (sivanantha321) [reply python/sklearnserver/pyproject.toml:3]: Done

**Comment by ckadner** [comment]
Hi @sivanantha321 I see the `Python package / build (3.11) (pull_request) Cancelled after 360m`. Was that on purpose? Are you making more changes?

**Comment by ckadner** [comment]
> Hi @sivanantha321 I see the `Python package / build (3.11) (pull_request) Cancelled after 360m`. Was that on purpose? Are you making more changes?

I see @yuzisun kicked it off again. I just cancelled it after more than one hour. Seems like both times it got stuck on `kserve/test/test_dataplane.py`?

**Comment by yuzisun** [comment]
> > Hi @sivanantha321 I see the `Python package / build (3.11) (pull_request) Cancelled after 360m`. Was that on purpose? Are you making more changes?
> 
> I see @yuzisun kicked it off again. I just cancelled it after more than one hour. Seems like both times it got stuck on `kserve/test/test_dataplane.py`?

yes not sure why, need to try replicating locally

**Comment by ckadner** [comment]
Would it make sense to add some timeouts? Might be tricky with `pytest` and `asyncio`, but there maybe some workarounds worth exploring https://stackoverflow.com/questions/55684737/how-to-timeout-an-async-test-in-pytest-with-fixture

**Comment by yuzisun** [comment]
> Would it make sense to add some timeouts? Might be tricky with `pytest` and `asyncio`, but there maybe some workarounds worth exploring https://stackoverflow.com/questions/55684737/how-to-timeout-an-async-test-in-pytest-with-fixture

These are unit tests which makes no sense to timeout, they normally run pretty quick something is definitely wrong.

**Comment by ckadner** [comment]
The `test_dataplane` and `test_server` both use `ray.serve`. Could it be due to Ray/Serve support being experimental on Python 3.11?

> **Note**: Python 3.11 support is experimental.

I find several issues related to Python 3.11 and Ray, like this one: https://github.com/ray-project/ray/issues/31606

**Comment by ckadner** [comment]
The `test_dataplane` unit tests fail with Ray `2.6.1` in this PR:
https://github.com/kserve/kserve/actions/runs/5668424182/job/15372037186?pr=3040#step:7:129

> ```
>   • Installing ray (2.6.1)
> ...
> kserve/kserve/storage/test/test_azure_storage.py ..........              [  5%]
> kserve/kserve/storage/test/test_s3_storage.py .....                      [  7%]
> kserve/kserve/storage/test/test_storage.py ............................. [ 22%]
> ...                                                                      [ 23%]
> kserve/test/test_creds_utils.py .........                                [ 28%]
> Error: The operation was canceled.
> ```

The `test_dataplane` unit tests passes when Ray `2.4.0` is getting installed (I started the Python package build on master):
https://github.com/kserve/kserve/actions/runs/5675244300/job/15380189347#step:7:129

```
>   • Installing ray (2.4.0)
> ...
> Test kserve:

> kserve/kserve/storage/test/test_azure_storage.py ..........              [  5%]
> kserve/kserve/storage/test/test_s3_storage.py .....                      [  7%]
> kserve/kserve/storage/test/test_storage.py ............................. [ 22%]
> ...                                                                      [ 23%]
> kserve/test/test_creds_utils.py .........                                [ 28%]
> kserve/test/test_dataplane.py .....................                      [ 39%]
> kserve/test/test_inference_service_client.py ......                      [ 42%]
> kserve/test/test_knative_addressable.py .                                [ 42%]
> kserve/test/test_knative_condition.py .                                  [ 43%]
> kserve/test/test_knative_url.py .                                        [ 43%]
> kserve/test/test_knative_volatile_time.py .                              [ 44%]
> kserve/test/test_kubeconfig_dict.py ..                                   [ 45%]
> kserve/test/test_model_repository_extension.py ......                    [ 48%]
> kserve/test/test_net_url_userinfo.py .                                   [ 48%]
> (ServeReplica:DummyServeModel pid=2527) INFO 2023-07-27 00:56:36,760 DummyServeModel DummyServeModel#MnPrdX replica.py:527 - LOAD OK 0.6ms
> kserve/test/test_server.py ...................................           [ 66%]
> kserve/test/test_v1alpha1_built_in_adapter.py .                          [ 67%]
> kserve/test/test_v1alpha1_cluster_serving_runtime.py .                   [ 67%]
...
```

File: python/kserve/poetry.lock [hunk 2]
```diff
@@ -2633,10 +2668,91 @@ serve = ["aiohttp (>=3.7)", "aiohttp-cors", "aiorwlock", "colorful", "fastapi",
 train = ["pandas", "requests", "tabulate", "tensorboardX (>=1.9)"]
 tune = ["pandas", "requests", "tabulate", "tensorboardX (>=1.9)"]
 
+[[package]]
+name = "ray"
+version = "2.6.1"
```
Comment (ckadner) [inline python/kserve/poetry.lock]: can you temporarily revert/pin `ray` to version `2.4.0` to find out if that will allow the `test_dataplane.py` unit test to pass?

**Review by ckadner — CHANGES_REQUESTED** [review]

File: python/kserve/poetry.lock (hunk 2 above)
Comment (sivanantha321) [reply python/kserve/poetry.lock]: Done

Comment (sivanantha321) [reply python/kserve/poetry.lock]: The tests are passing with `ray` version `2.4.0`.

**Comment by yuzisun** [comment]
Thanks @sivanantha321 !

/lgtm
/approve

File: python/sklearnserver/pyproject.toml (hunk 1 above)
Comment (rachitchauhan43) [reply python/sklearnserver/pyproject.toml:3]: @sivanantha321 : Which script is that ? We would need to make it part of Release process if it is not yet.
cc: @yuzisun

Comment (yuzisun) [reply python/sklearnserver/pyproject.toml:3]: it is added to the release process, see step 4
https://github.com/kserve/kserve/blob/master/release/RELEASE_PROCESS_v2.md#on-the-release-day

**Pull request merged**